creation/interfaces-and-networks.md

```diff
@@ -548,64 +548,52 @@ the interface should be configured as follows.
 
-#### masquerade - IPv6 support
-It is currently experimental, but `masquerade` mode can be used in IPv6 clusters
-(not dual stack - IPv6 only).
+#### masquerade - IPv4 and IPv6 dual-stack support
+
+It is currently experimental, but `masquerade` mode can be used in IPv4 and IPv6
+dual-stack clusters to provide a VM with an IP connectivity over both protocols.
 
 As with the IPv4 `masquerade` mode, the VM can be contacted using the pod's IP
-address - which will be an IPv6 one. Outgoing traffic is also "NAT'ed" to the
-pod's IPv6 address.
+address - which will be in this case two IP addresses, one IPv4 and one
+IPv6. Outgoing traffic is also "NAT'ed" to the pod's respective IP address
+from the given family.
 
-Unlike in IPv4, the configuration of the IP address, default route, and DNS is
+Unlike in IPv4, the configuration of the IPv6 address and the default route is
 not automatic; it should be configured via cloud init, as shown below:
 
-<pre>
-    kind: VM
-    metadata:
-      namespace: <i>your-favorite-namespace</i>
-    spec:
-      domain:
-        devices:
-          disks:
-            - disk:
-              bus: virtio
-              name: cloudinitdisk
-          interfaces:
-            - name: red
-              masquerade: {} # connect using masquerade mode
-              ports:
-                - port: 80 # allow incoming traffic on port 80 to get into the virtual machine
-      networks:
-      - name: red
-        pod: {}
-      volumes:
-      - cloudInitNoCloud:
-          networkData: |
-            version: 2
-            ethernets:
-              eth0:
-                addresses: [ fd10:0:2::2/120 ]
-                gateway6: fd10:0:2::1
-                nameservers:
-                  addresses: [ <i>dns_server_ip</i> ]
-                  search:
-                    - <i>your-favorite-namespace</i>.svc.cluster.local
-                    - svc.cluster.local
-                    - cluster.local
-          userData: |-
-            #!/bin/bash
-            echo "fedora" |passwd fedora --stdin
-</pre>
-
-> **Note:** The IP address for the VM and default gateway **must** be the ones
-> shown above.
-
-> **Note:** DNS must also be manually configured. To know which DNS server to
-> use, run the following query:
-> `kubectl get service -n kube-system kube-dns -o=custom-columns=IP:.spec.clusterIP`
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      disks:
+        - disk:
+          bus: virtio
+          name: cloudinitdisk
+      interfaces:
+        - name: red
+          masquerade: {} # connect using masquerade mode
+          ports:
+            - port: 80 # allow incoming traffic on port 80 to get into the virtual machine
+  networks:
+  - name: red
+    pod: {}
+  volumes:
+  - cloudInitNoCloud:
+      networkData: |
+        version: 2
+        ethernets:
+          eth0:
+            dhcp4: true
+            addresses: [ fd10:0:2::2/120 ]
+            gateway6: fd10:0:2::1
```

Comment:
did you check it on single stack ipv6  / ipv4 in case needed ?

well on single stack ipv4 it wont work i guess, unless you dont want to have document for single stack anymore

Reply:
This section is for dual-stack. The default supported single-stack is documented in the `masquerade` section above. The dual-stack subsection is a "note" for this experimental feature and may be dropped once it's seamlessly supported. I hope this answers the questions you posted above too.

Reply:
Thanks